### ISSUE TITLE
Update caliper and improve output

### ISSUE THREAD

**Issue by wrtobin** [GEOS-DEV/GEOS#1084]
- [x] We need to update caliper to include a fix related to the time history output. 
- [x] Additionally when we update we want to rebuild with improved column output support.
- [ ] Also we want to modify the mark function macro in order to discriminate between overloaded functions, and include template parameter information where appropriate.

**Comment by wrtobin** [comment]
Caliper was updated in the TPLs prior to the initial time-hist functionality merge, but I don't think the caliper function-marking macro to account for overloaded functions and multiple instantiations of the same template function as caliper scope symbols are currently either user-provided strings or function names (sans parameters/template parameters) with namespace scoping included.

**Comment by wrtobin** [comment]
While working on this it would probably be good to re-instrument the packing functions now that they're largely async, in order to more accurately reflect their performance, as the current instrumentation only captures the unpacking performance adequately.

**Comment by TotoGaz** [comment]
@wrtobin Do you think that the TPL upgrade of `caliper` in https://github.com/GEOS-DEV/thirdPartyLibs/pull/200 solves the `caliper` part of the issue?

**Comment by wrtobin** [comment]
Yeah the first 2 ticks can be resolved, especially since for critical profiling I usually retain the full caliper database now instead of just the stdout report. Don't think our mark macro discriminates between overloaded functions yet, not least that I recall.

**Comment by paveltomin** [comment]
@wrtobin any update?
